src/langsmith/diagnostics-self-hosted.mdx

```diff
@@ -0,0 +1,184 @@
+- **`langsmith-operator`**: The `lgps` Kubernetes operator that reconciles underlying Kubernetes resources for `lgps` resources. Part of the data plane infrastructure.
+
+<Note>
+This list is not exhaustive. Additional components may be present in your deployment depending on your configuration. For a complete overview, refer to [LangSmith Deployment components](/langsmith/components).
+</Note>
+
+### Review application logs
+
+Tail the logs of each pod to understand baseline behavior:
+
+```bash
+kubectl logs -f <pod_name>
+```
+
+#### What to look for in logs
+
+First, enable `DEBUG` level logs (see Step 2 below). Then look for these log lines:
+
+- **`langsmith-listener`**: `Reconciling projects...` (appears every 10 seconds)
+- **`langsmith-operator`**: `Starting reconciliation` (appears periodically)
+
+In a healthy deployment, you should not see any errors. All logs should appear normal and routine.
+
+### Next steps
+
+Once you understand the baseline state, create a deployment from the LangSmith Frontend. This process will create an `lgps` Kubernetes resource that you can monitor.
+
+## Step 2. Enable debug logging
+
+When troubleshooting issues, the first step is typically to enable debug-level logging to gather more detailed information about what's happening in your system.
+
+### For control plane or data plane deployments
+
+If you are experiencing issues with a control plane deployment (for example, `langsmith-host-backend`) or a data plane deployment (for example, `langsmith-listener`), reinstall the Helm chart with the `LOG_LEVEL=DEBUG` environment variable. Add the following to your `values.yaml` file:
+
+```yaml
+extraEnv:
+  - name: LOG_LEVEL
+    value: DEBUG
+```
+
+### For Agent Server deployments
+
+If the issue is with an individual Agent Server deployment:
+
+1. Navigate to the **Deployments** tab in the [LangSmith UI](https://smith.langchain.com).
+1. On a deployment's view, select **+ New Revision**.
+1. Set the **Environment Variables** to `LOG_LEVEL` and the value to `DEBUG`.
+
+<Note>
+You can also find debug logs in the UI on a deployment's view, click on **Server Logs** and select **Debug** for the **Log level: Info** dropdown.
+</Note>
+
+### For widespread issues
+
+If you are unsure where the issue originates, enable `DEBUG` logging everywhere (control plane, data plane, and all Agent Server deployments).
+
+### Interpret debug logs
+
+Once you have reviewed logs when the system was working correctly to establish what normal behavior looks like, look for the following problem indicators:
+
+- Exceptions or stack traces.
+- Error messages (the word `"ERROR"` or red-colored text).
+- Unusual patterns that differ from normal operation.
+
+Based on the errors you find:
+
+- **Infrastructure bug**: If you suspect a bug in the platform infrastructure, raise the issue with LangChain.
```

Comment:
Suggestion/preference: Let's exclude this bullet point. My preference is to not normalize the existence of bugs in the platform.